sqlglot/dialects/dialect.py

```diff
@@ -220,14 +220,14 @@ def tokenize(self, sql: str) -> t.List[Token]:
     @property
     def tokenizer(self) -> Tokenizer:
         if not hasattr(self, "_tokenizer"):
-            self._tokenizer = t.cast(t.Type[Tokenizer], self.tokenizer_class)()
+            self._tokenizer = self.tokenizer_class()
         return self._tokenizer
 
     def parser(self, **opts) -> Parser:
-        return t.cast(t.Type[Parser], self.parser_class)(**opts)
+        return self.parser_class(**opts)
 
     def generator(self, **opts) -> Generator:
-        return t.cast(t.Type[Generator], self.generator_class)(**opts)
+        return self.generator_class(**opts)
```

Comment:
Should we get rid of `parser` and `generator` and just rename the internal fields (*_class) to these instead? Perhaps do the same for the `tokenizer`?

cc: @tobymao

Reply:
FYI: I've made it so that the internal `*_class` fields of `Dialect` are initialized to the corresponding classes instead of `None`.

Reply:
not sure what you mea

Reply:
I mean get rid of the `parser` method and rename `parser_class` to `parser` (and similarly got the `generator` method).

Reply:
sure

Reply:
Nvm I think it doesn't hurt to leave it as is. Tried to make the change and it felt kinda asymmetric because we don't do the same for the tokenizer (since we cache it), so it looks weird. I'll just go ahead and merge without this.